src/main/java/com/bruincreates/server/service/ProductReviewService.java

```diff
@@ -1,13 +1,104 @@
 package com.bruincreates.server.service;
 
 import com.bruincreates.server.dao.mapper.ProductReviewMapper;
+import com.bruincreates.server.dao.po.ProductReview;
+import com.bruincreates.server.dao.po.ProductReviewExample;
+import com.bruincreates.server.model.request.CreateReviewRequest;
+import com.bruincreates.server.model.request.DeleteReviewRequest;
+import com.bruincreates.server.exception.BadRequestException;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
 @Service
 public class ProductReviewService {
 
     @Autowired
     ProductReviewMapper productReviewMapper;
 
+    @Autowired
+    OrderService orderService;
+
+    public void createReview (CreateReviewRequest request) throws BadRequestException {
+
+        String username = request.getUsername();
+        String productId = request.getProductId();
+        String reviewContent = request.getProductReview();
+
+        // Validate reviewer has purchased product
+        if(!orderService.hasCompletedOrder(username, productId)){
+            throw new BadRequestException("user has not completed an order of this product");
+        }
+
+        // Validate reviewer did not previously review this product
+        ProductReviewExample reviewExample = new ProductReviewExample();
+        reviewExample.createCriteria().andBuyerIdEqualTo(username).andProductIdEqualTo(productId);
+        List<ProductReview> reviews = productReviewMapper.selectByExample(reviewExample);
+
+        int reviewsLength = reviews.size();
+        if(reviewsLength == 1){
+            throw new BadRequestException("user has already made a review for this product");
+        }
+        else if(reviewsLength > 1) {
+            throw new BadRequestException("database needs fixing: there is more than 1 review in the db by this user for this product");
+        }
```

Comment:
I personally prefer to refractor the above part into a separate function canReview, so we can later add a canReview API for the frontend to gray out review buttons for customers who cannot review a product.